src/commonTest/kotlin/fr/acinq/bitcoin/DeterministicWalletTestsCommon.kt

```diff
@@ -307,10 +308,23 @@ class DeterministicWalletTestsCommon {
 
     @Test
     fun `parse string-formatted derivation paths`() {
-        assertEquals(KeyPath("m/44'/0'/0'/0"), KeyPath(listOf(DeterministicWallet.hardened(44), DeterministicWallet.hardened(0), DeterministicWallet.hardened(0), 0)))
-        assertEquals(KeyPath("/44'/0'/0'/0"), KeyPath(listOf(DeterministicWallet.hardened(44), DeterministicWallet.hardened(0), DeterministicWallet.hardened(0), 0)))
-        assertEquals(KeyPath("44'/0'/0'/0"), KeyPath(listOf(DeterministicWallet.hardened(44), DeterministicWallet.hardened(0), DeterministicWallet.hardened(0), 0)))
-        assertEquals(KeyPath("m/44/0'/0'/0"), KeyPath(listOf(44, DeterministicWallet.hardened(0), DeterministicWallet.hardened(0), 0)))
+        assertEquals(KeyPath("m/44'/0'/0'/0"), KeyPath(listOf(hardened(44), hardened(0), hardened(0), 0)))
+        assertEquals(KeyPath("m/44h/0h/0h/0"), KeyPath(listOf(hardened(44), hardened(0), hardened(0), 0)))
+
+        assertEquals(KeyPath("m/44'/0'/0'/0").asString('\''), "m/44'/0'/0'/0")
+        assertEquals(KeyPath("m/44'/0'/0'/0").asString('h'), "m/44h/0h/0h/0")
+        assertEquals(KeyPath("m/44'/0'/0'/0").asString('h'), "m/44h/0h/0h/0")
```

Comment:
nit: this is a duplicate of the previous line, I believe you wanted to do:

```suggestion
        assertEquals(KeyPath("m/44h/0h/0h/0").asString('h'), "m/44h/0h/0h/0")
```

Reply:
thanks, fixed in https://github.com/ACINQ/bitcoin-kmp/commit/e2f1642bd5993f2bb1ae1017c294fa1b3fa4c6d0